Lib/ufo2ft/util.py

```diff
@@ -612,3 +612,15 @@ def unicodeScriptExtensions(
     defines "Hrkt" as an alias for both scripts.
     """
     return {aliases.get(s, s) for s in unicodedata.script_extension(chr(codepoint))}
+
+
+def describe_ufo(ufo: Any) -> str:
+    """Returns a description of a UFO suitable for logging."""
+    if (
+        hasattr(ufo, "reader")
+        and hasattr(ufo.reader, "fs")
+        and hasattr(ufo.reader.fs, "root_path")
+    ):
+        return ufo.reader.fs.root_path
+    else:
+        return f"{ufo.info.familyName} {ufo.info.styleName}"
```

Comment:
these can both be unset or None, maybe if you end up with an empty string just use its repr(ufo)